sklearn/decomposition/dict_learning.py

```diff
@@ -343,35 +352,32 @@ def _update_dict(dictionary, Y, code, verbose=False, return_r2=False,
         Updated dictionary.
 
     """
-    n_components = len(code)
-    n_samples = Y.shape[0]
+    n_features, n_components = B.shape
     random_state = check_random_state(random_state)
     # Residuals, computed 'in-place' for efficiency
-    R = -np.dot(dictionary, code)
-    R += Y
+    R = -np.dot(dictionary, A)
+    R += B
     R = np.asfortranarray(R)
-    ger, = linalg.get_blas_funcs(('ger',), (dictionary, code))
+    ger, = linalg.get_blas_funcs(('ger',), (dictionary, A))
     for k in range(n_components):
         # R <- 1.0 * U_k * V_k^T + R
-        R = ger(1.0, dictionary[:, k], code[k, :], a=R, overwrite_a=True)
-        dictionary[:, k] = np.dot(R, code[k, :].T)
+        R = ger(1.0, dictionary[:, k], A[k, :], a=R, overwrite_a=True)
+        dictionary[:, k] = np.dot(R, A[k, :])
```

Comment:
Hm, I'd expect things to be numerically a bit different due to at least one of the following

- the modified projection formula
- the indentation issue on the second call to `ger(...)`

I'd have proposed to open a separate issue on these things and then address them in a separate PR, but then this would be a really minor change (in terms of patch size). So i'd rather address them as part of this PR, except there are compelling reasons to do otherwise. What do you think ?